### ISSUE TITLE
Move desktop target platform artifacts out of host platform artifact directory.

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#38935]
The debug desktop target platform is currently cohabitation with the host platform artifacts of the same name. The debug target platform artifacts should be moved to `platform-debug` to match the reset of the artifact repository.

This could be done via a soft transition by upload the artifacts to both directories, and then removing the old upload once the tool is updated.

**Comment by stuartmorgan** [comment]
https://flutter-review.googlesource.com/c/recipes/+/3220

**Comment by cbracken** [comment]
Deduped against #108756, which I'm working on right now.

**Comment by bkonyi** [comment]
@cbracken is this still a valid issue?

**Comment by cbracken** [comment]
Yes. We should really not be building the desktop-specifc products like FlutterMacOS.framework or the Windows DLL as part of the host build. The host build should be host products that are used regardless of target -- e.g. the Dart SDK, flutter_tester, etc.

**Comment by jmagman** [comment]
> We should really not be building the desktop-specifc products like FlutterMacOS.framework or the Windows DLL as part of the host build. The host build should be host products that are used regardless of target -- e.g. the Dart SDK, flutter_tester, etc.

I reopened https://github.com/flutter/flutter/issues/108756 to track the engine build rule work.
